common/djangoapps/third_party_auth/tests/test_pipeline_integration.py

```diff
@@ -208,3 +209,75 @@ def test_for_value_error_if_provider_id_invalid(self):
 
         with self.assertRaises(ValueError):
             pipeline.get_complete_url(provider_id)
+
+
+@unittest.skipUnless(
+    testutil.AUTH_FEATURES_KEY in settings.FEATURES, testutil.AUTH_FEATURES_KEY + ' not in settings.FEATURES')
+class TestPipelineUtilityFunctions(TestCase, test.TestCase):
+    """
+    Test some of the isolated utility functions in the pipeline
+    """
+    def setUp(self):
+        super(TestPipelineUtilityFunctions, self).setUp()
+        self.user = social_models.DjangoStorage.user.create_user(username='username', password='password')
+        self.social_auth = social_models.UserSocialAuth.objects.create(
+            user=self.user,
+            uid='fake uid',
+            provider='fake provider'
+        )
+
+    def test_get_real_social_auth_from_dict(self):
+        request = mock.MagicMock(
+            session={
+                'partial_pipeline': {
+                    'kwargs': {
+                        'social': {
+                            'uid': 'fake uid'
+                        }
+                    }
+                }
+            }
+        )
+        real_social = pipeline.get_real_social_auth_object(request)
+        self.assertEqual(real_social, self.social_auth)
+
+    def test_get_real_social_auth(self):
+        request = mock.MagicMock(
+            session={
+                'partial_pipeline': {
+                    'kwargs': {
+                        'social': self.social_auth
+                    }
+                }
+            }
+        )
+        real_social = pipeline.get_real_social_auth_object(request)
+        self.assertEqual(real_social, self.social_auth)
+
+    def test_get_real_social_auth_no_pipeline(self):
+        request = mock.MagicMock(session={})
+        real_social = pipeline.get_real_social_auth_object(request)
+        self.assertEqual(real_social, None)
+
+    def test_get_real_social_auth_no_social(self):
```

Comment:
Kindly add docstring that represent the test functionality. it always helpful to understand.